Userland/Libraries/LibJS/Runtime/ErrorTypes.h

```diff
@@ -110,6 +110,7 @@
         "Object prototype must not be {} on a super property access")                                                                   \
     M(ObjectPrototypeWrongType, "Prototype must be an object or null")                                                                  \
     M(OptionIsNotValidValue, "{} is not a valid value for option {}")                                                                   \
+    M(OutOfMemory, "Out of memory")                                                                                                     \
```

Comment:
I liked the error that shows the amount (attempted to be) allocated, is there any situation where we won't know that? Perhaps `TRY_OR_THROW_OOM` could always require a number of bytes as the third parameter and then we wouldn't need to have two errors for the same thing.

Reply:
or perhaps it should be propagated as part of the `ENOMEM` `ErrorOr<T>`...

Reply:
I was kinda on the fence here. Wrapping `DeprecatedString::format` is one case where we don't know what the final size will be. We also don't know how big UTF-8 to/from UTF-16 conversions will end up unless we wanna loop over the source string a second time.